include/uriparser/UriBase.h

```diff
@@ -97,6 +97,25 @@
 
 
 
+/* Import/export decorator */
+#if defined(URI_LIBRARY_BUILD)
+# if defined(_MSC_VER)
+#  define URI_PUBLIC __declspec(dllexport)
+# elif defined(URI_VISIBILITY)
+#  define URI_PUBLIC __attribute__ ((visibility("default")))
+# else
+#  define URI_PUBLIC
+# endif
+#else
+# if defined(_MSC_VER)
+#  define URI_PUBLIC __declspec(dllimport)
+# else
+#  define URI_PUBLIC
+# endif
+#endif
+
```

Comment:
To make static lib build work as well, an additional is needed:
```c
#if defined(URI_STATIC_LIB)
# define URI_PUBLIC
#else
# if defined(URI_LIBRARY_BUILD)
#  if defined(_MSC_VER)
#   define URI_PUBLIC __declspec(dllexport)
#  elif defined(URI_VISIBILITY)
#   define URI_PUBLIC __attribute__ ((visibility("default")))
#  else
#   define URI_PUBLIC
#  endif
# else
#  if defined(_MSC_VER)
#   define URI_PUBLIC __declspec(dllimport)
#  else
#   define URI_PUBLIC
#  endif
# endif
#endif
```
Then, these lines should be added to `CMakeLists.txt`:
```cmake
if (NOT BUILD_SHARED_LIBS)
    target_compile_definitions(uriparser PUBLIC URI_STATIC_LIB)
endif ()
```

Reply:
If we don't do like this, there will be some link errors:
>3>uriparse.obj : error LNK2019: unresolved external symbol __imp__uriParseUriA referenced in function _main
3>uriparse.obj : error LNK2019: unresolved external symbol __imp__uriFreeUriMembersA referenced in function _main
3>C:\Users\vcpkg\Desktop\uriparser\build\Debug\uriparse.exe : fatal error LNK1120: 2 unresolved externals

Reply:
I think I have both dynamic and static building fixed now. Thanks a lot for your testing and ideas!